walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -411,31 +413,75 @@ public void allOptionalAreIgnored() {
 
     Collection<TimeRange> actual = query.query(events, request);
     Collection<TimeRange> expected =
-        Arrays.asList(TimeRange.fromStartEnd(TIME_0830AM, TimeRange.END_OF_DAY, true));
+        Arrays.asList(TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false));
 
     Assert.assertEquals(expected, actual);
   }
 
   @Test
-  public void eventWithRequiredAndOptionalAttendees() {
-    // An event with both an optional and required attendee is present. Even though the
-    // optional attendee is not considered due to other conflicts, this event should still
-    // influence the time ranges.
+  public void optionalOptimizationForQuantity() {
+    // Three optional attendees are present. Although only considering one optional attendee
+    // would result in more available time ranges, two optional attendees can be considered in
+    // another optimization. The selection with the most optional attendees should be 
+    // preferred.
+
+    // Required:   |---|  |-|   |--|
+    // Optional 1:     |--|
+    // Optional 2:          |---|
+    // Optional 3:       |------|
+    //
+    // Correct:    |---| |---------| (Optional 2 and 3)
 
     Collection<Event> events = Arrays.asList(
-        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
-            Arrays.asList(PERSON_A, PERSON_C)),
-        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
             Arrays.asList(PERSON_A)),
-        new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, DURATION_15_MINUTES),
-            Arrays.asList(PERSON_C)));
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 4", TimeRange.fromStartEnd(TIME_0800AM, TIME_0900AM, false),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_0830AM, TIME_1100AM, false),
+            Arrays.asList(PERSON_C)),
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_0930AM, TIME_1100AM, false),
+            Arrays.asList(PERSON_D)));
```

Comment:
```suggestion
        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
            Arrays.asList(PERSON_A)),
        new Event("Event 3", TimeRange.fromStartEnd(TIME_0930AM, TIME_1100AM, false),
            Arrays.asList(PERSON_A, PERSON_C, PERSON_D)));
        new Event("Event 4", TimeRange.fromStartEnd(TIME_0800AM, TIME_0900AM, false),
            Arrays.asList(PERSON_B)),
```

Reply:
Would you be able to explain your thought process behind this change? I understand consolidating `PERSON_C` and `PERSON_D` into a single event, but adding `PERSON_A` to `Event 3` and removing the event from 11AM to the end of the day changes the nature of the test. Of the aspects changed it most importantly means that all optional attendees can now attend instead of either `PERSON_B` or `PERSON_C` and `PERSON_D` as before.

Reply:
I have a feeling this is likely due to my commenting not being great for the nature of this test. I will take another look at that.

Reply:
Also writing it here to make sure you see it -- sorry for making you go through reading those. I'll definitely be more mindful of existing conventions in the future.

Reply:
I updated the comments for this test and the others. This might help to clarify what the intention of the test is.

Reply:
Oh, I didn't catch that event 3 was different for `PERSON_A`, `PERSON_B` and `PERSON_C`. If they are separate events give them unique names if possible. It's confusing to me if there are three distinct events and they are all called "event 3"

Reply:
Yeah that was an oversight on my part, sorry. I changed that in the most recent update.